problem_set_1.py

```diff
@@ -1,38 +1,56 @@
 # 1. Set the variable `given_name` to the string "Addison".
-
+given_name = "Addison"
 
 # ------------------------------------------------------------------------------
 # 2. You have 20 candies that you must divide equally among 6 people. How many candies will be left over?
 # Set variables for `candies`, `people`, `left_over` to make your tests pass.
-
+candies = 20
+people = 6
+left_over = candies % people
 
 # ------------------------------------------------------------------------------
 # 3. Create a function called `greeting` that returns "Hello, <name>!",
 # where <name> is the name given as an argument to the function.
+def greeting(name):
+    return "Hello, " + name + "!"
 
 
 # ------------------------------------------------------------------------------
 # 4. Create a function called `is_odd` that, given a number, will
 # return true if the number is odd and false if it is not. An odd number is a
 # number which, when divided by 2, has a remainder of 1 or -1.
-
+def is_odd(number):
+    if number % 2 == 1 or number % 2 == -1:
+        return True
+    else:
+        return False
 
 # ------------------------------------------------------------------------------
 # 5. Create a function called `is_even` that, given a number, will
 # return true if the number is even and false if it is not. An even number is a
 # number which, when divided by 2, has a remainder of 0.
+def is_even(number):
+    if number % 2 == 0:
+        return True
+    else:
+        return False
 
```

Comment:
Given that this is in the same file and comes after `is_odd()`, is there a way to use that function to write this one?

Reply:
Oh man, this was not as easy as I thought it would be--I tried a few ways of using is_odd (including `return not is_odd(number)`) but all of them seemed to cause errors for tests run for `gcd()`. My most recent attempt before I decided to move on is seen on line 44 here. Might come back to this later.